latex comment filter

MartinoMensio opened the pull request:
This filter excludes comments (starting with `%`).

This matches:
- line entirely commented out: `% this is a comment`
- line with ending part commented: `this is text and %this is a comment`

And does not match:
- escaped `%` sign: `this is 30\%, and is not a comment`

How it works:
The regex `/(?<!\\)%.*/g` has:
- negative lookbehind: the character before must be different from `\`
- the percentage sign
- followed by any characters (until newline, where comment ends)

This filter has been tested by using the user filters in the extension. Please let me know if it has some issues

robindijkhof commented:
Hi, I'm not entirely sure what problem this PR solves. AFAIK comments are already filtered and escaped % are still displayed.

![image](https://user-images.githubusercontent.com/22224763/109027456-194a5680-76c1-11eb-884e-760a6faa8675.png)

MartinoMensio commented:
Hi,
Sorry, you are right. It is already filtered by your first filter `/w*(?<!\\)%.*\n?/g`.

I didn't see it working because I have some errors in my console for my file:

```text
2021-02-24T16:12:02.256Z Pattern incorrect. Text length reduced by :/w*(?<!\\)%.*\n?/g,
```

This is probably caused by the last line of my tex file that is currently a comment. 
So when you check on the number of lines (`newText.split("\n").length === text.split("\n").length`) there is something not working.

The minimal file to reproduce the issue is the following:
```
a
%b
```

Without a new line at the end.

So the fix I propose is now the following:
- remove my added pattern because it's a duplicate
- correct your first pattern to `{ pattern: /w*(?<!\\)%.*(\n?)/g, newValue: '$1' },` where the optional `\n` is then put as newValue

I hope this helps

robindijkhof commented:
Thanks!
